src/azure-cli-core/azure/cli/core/tests/test_profile.py

```diff
@@ -716,6 +716,17 @@ def test_get_raw_token(self, mock_get_token, mock_read_cred_file):
         self.assertEqual(sub, '1')
         self.assertEqual(tenant, self.tenant_id)
 
+        # Test get_raw_token with tenant
+        creds, sub, tenant = profile.get_raw_token(resource='https://foo', tenant=self.tenant_id)
+
+        self.assertEqual(creds[0], self.token_entry1['tokenType'])
```

Comment:
Can we also add tests for MSI and cloud shell? (check exception)

Reply:
Yes. It is already in this file.